app/assets/javascripts/modules/want_to_talk.js

```diff
@@ -0,0 +1,244 @@
+        Math.max(document.documentElement.clientWidth, w.innerWidth || 0) >= MEDIUM_MIN_WIDTH);
+    }
+
+    function showElement() {
+      if (!isInSidebar && isMQMedium()) {
+        showInSidebar();
+        isInSidebar = true;
+        return;
+      }
+
+      if (isInSidebar && !isMQMedium()) {
+        showInline();
+        isInSidebar = false;
+        return;
+      }
+    }
+
+    function positionComponent() {
+      var scrollTop;
+
+      if (!isInSidebar) {
+        return;
+      }
+
+      scrollTop = $(w).scrollTop();
+
+      if (!isFixed && !isAtBottom && scrollTop > top) {
+        $sidebarEl.addClass(FIXED_CLASS);
+        isFixed = true;
+        return;
+      }
+
+      if (isFixed) {
+        if (scrollTop < top) {
+          $sidebarEl.removeClass(FIXED_CLASS);
+          isFixed = false;
+          return;
+        }
+
+        console.info(scrollTop + '=' + bottom);
+        if (scrollTop > bottom) {
+          $sidebarEl
+            .removeClass(FIXED_CLASS)
+            .addClass(BOTTOM_CLASS);
+          isFixed = false;
+          isAtBottom = true;
+          return;
+        }
+      }
+
+      if (isAtBottom && scrollTop < bottom) {
+          $sidebarEl
+            .addClass(FIXED_CLASS)
+            .removeClass(BOTTOM_CLASS);
+        isAtBottom = false;
+      }
+    }
+
+    function handleResize() {
+      measureDOM();
+      showElement();
+      positionComponent();
+    }
+
+    function handleScroll() {
+      if (!isInSidebar) {
+        return;
+      }
+
+      positionComponent();
+    }
+
+    function track(action, label) {
+      MAS.publish('analytics:trigger', {
+        event: 'gaEvent',
+        gaEventCat: TRACK_CATEGORY,
+        gaEventAct: action,
+        gaEventLab: label
+      });
+    }
+
+    function debounce(func, wait) {
+      var timeout;
+
+      return function() {
+        var context = this, args = arguments,
+            later = function () {
+              timeout = null;
+              func.apply(context, args);
+            };
+        clearTimeout(timeout);
+        timeout = setTimeout(later, wait);
+      };
+    }
+
+    function handleTriggerClick(e) {
+      var $el = $(e.target).closest(COMPONENT_SELECTOR),
+          $trigger = $el.find(TRIGGER_SELECTOR),
+          $label = $el.find(LABEL_SELECTOR),
+          $body = $el.find(BODY_SELECTOR);
+
+      if ($el.hasClass(OPEN_CLASS)) {
+        $el.removeClass(OPEN_CLASS);
+        $trigger.attr(ARIA_EXPANDED, 'false');
+        $label.text($el.attr('data-open') || I18N_OPEN);
+        $body.attr(ARIA_HIDDEN, 'true');
+        track('Click', 'Close');
+      } else {
+        $el.addClass(OPEN_CLASS);
+        $trigger.attr(ARIA_EXPANDED, 'true');
+        $label.text($el.attr('data-close') || I18N_CLOSE);
+        $body.attr(ARIA_HIDDEN, 'false');
+        track('Click', 'Open');
+      }
+
+      if ($el.hasClass(SIDEBAR_CLASS)) {
+        // wait for CSS transition
+        setTimeout(function() {
```

Comment:
I've had issues with onTransitionEnd not firing if the tab in focus when the animation ends.